WalletWasabi.Tests/UnitTests/Crypto/ProofSystemTests.cs

```diff
@@ -41,6 +44,76 @@ public void CanProveAndVerifyMAC()
 			var clientStatement = ProofSystem.IssuerParameters(coordinatorParameters, mac, Ma);
 			var isValidProof = ProofSystem.Verify(clientStatement, proofOfMac);
 
+			Assert.True(isValidProof);
+			////////
+
+			var corruptedResponses = new ScalarVector(proofOfMac.Responses.Reverse());
+			var invalidProofOfMac = new Proof(proofOfMac.PublicNonces, corruptedResponses);
+			isValidProof = ProofSystem.Verify(clientStatement, invalidProofOfMac);
+
+			Assert.False(isValidProof);
+
+			var corruptedPublicNonces = new GroupElementVector(proofOfMac.PublicNonces.Reverse());
+			invalidProofOfMac = new Proof(corruptedPublicNonces, proofOfMac.Responses);
+			isValidProof = ProofSystem.Verify(clientStatement, invalidProofOfMac);
+
+			Assert.False(isValidProof);
+		}
+
+		[Fact]
+		[Trait("UnitTest", "UnitTest")]
+		public void CanProveAndVerifySerialNumbers()
+		{
+			var rnd = new SecureRandom();
+			var z = rnd.GetScalar();
+			var r = rnd.GetScalar();
+			var a = rnd.GetScalar();
```

Comment:
add a case when `a = Scalar.Zero`? it doesn't really add anything to coverage, but is a realistic example

Reply:
Done.